src/plugins/udps/udps_cli.c

```diff
@@ -249,17 +252,22 @@ void
 udps_dump_policy_rule_by_id (vlib_main_t * vm, u8* policy_name, u32 id) 
 {
   udps_rule_entry_t *re;
+  char delimiter[65] = "";
   bool ret = udps_db_rule_entry_get(policy_name, id, &re);
   if (!ret) {
-     vlib_cli_output(vm, "No policy rule found for given name and id");
+     vlib_cli_output(vm, "No policy rule found for given name:%s and id:%d", policy_name, id);
      return;
   }
-  vlib_cli_output(vm, "Policy name=%s  rule id=%u action_id=%u offset=%d value= ", 
-	               policy_name, re->rule_id,  re->act_id, re->match_pkt->offset);
-    
-  for (int i =0; i<vec_len(re->match_pkt->value);i++) {
-     vlib_cli_output(vm, "%x",re->match_pkt->value[i]);
-  } 
+  vlib_cli_output(vm, "Policy Rule");
+  for (int i = 0; i < 65; i++) {
+     strcat(delimiter, "-");
+  }
+  vlib_cli_output(vm, "%s\n", delimiter);
+  vlib_cli_output(vm,"%-40s : %s", "Policy name", policy_name);
+  vlib_cli_output(vm,"%-40s : %u", "Rule Id", re->rule_id);
+  vlib_cli_output(vm,"%-40s : %u", "Action Id", re->act_id);
+  vlib_cli_output(vm,"%-40s : %d", "Packet Offset", re->match_pkt->offset);
```

Comment:
There can be multiple matched in a rule.

Reply:
oh it is an vector, in that case, we will need to modify this API too :
bool udps_db_rule_entry_add(u8 *name, u8 id, u8 oper, u16 offset, u8 *value, u8 *aname);